### ISSUE TITLE
should debug=no be the default?

### ISSUE THREAD

**Issue by robertmeta** [mawww/kakoune#2701]
I believe it is time to swap the default state of Kakoune after a standard build.  Anecdotally, in various places there seems to be an uptick of performance concerns that often are alleviated with a debug=no build.  I think that more users are building Kakoune to use it/try it at this point than to dev/debug it.

An example of this was today in #kakoune there was a user on gentoo who was very surprised when the result of their ebuild was a debug build. 

>  There's nothing that suggests it would have been a debug build

I think as the common case will be use of Kakoune not dev/debug we should change the default.  Possibly add a convenience env var it checks for to always do debug builds for people who prefer that and don't want to do debug=yes everytime.

**Comment by Muffindrake** [comment]
I don't feel it's ever a good idea to stand in the way of someone wanting to try your software and forcing them to find three pieces of text in different places how to generate a build that will run fine.

For that gentoo user's case, a debug build was generated despite the +debug USE flag inherent to the build script not being set. A different build, where the only changes were to environment variables CFLAGS and CXXFLAGS, resulted in an optimized build - I'm equally puzzled as to why that broke at that time..

**Comment by mawww** [comment]
Yeah, I guess it makes sense to make release build the default now.
